ocpp/messages.py

```diff
@@ -182,6 +182,53 @@ def get_validator(
     # character. If no encoding is given, reading the schema would fail with:
     #
     #     Unexpected UTF-8 BOM (decode using utf-8-sig):
+    try:
+        with open(path, 'r', encoding='utf-8-sig') as f:
+            data = f.read()
+            validator = Draft4Validator(json.loads(data,
+                                                   parse_float=parse_float))
+            _validators[cache_key] = validator
+        return _validators[cache_key]
+    except FileNotFoundError:
+        return _get_backport_validator(
+            message_type_id,
+            action,
+            ocpp_version,
+            parse_float
+        )
+
+
+def _get_backport_validator(
+        message_type_id: int,
+        action: str,
+        ocpp_version: str,
+        parse_float: Callable = float
+) -> Draft4Validator:
+    """
+    Reads backported schema from disk.
+
+    OCPP 1.6 Improved Security specification has selected Requests and
+    Responses backported from 2.0 to 1.6.
+    """
+    if ocpp_version not in ["1.6"]:
+        raise ValueError
+
+    schemas_dir = 'v' + ocpp_version.replace('.', '')
+
+    schema_name = action
+    if message_type_id == MessageType.CallResult:
+        schema_name += 'Response'
+    elif message_type_id == MessageType.Call:
+        schema_name += 'Request'
+
+    cache_key = schema_name + '_' + ocpp_version
+    if cache_key in _validators:
+        return _validators[cache_key]
+
+    dir,  _ = os.path.split(os.path.realpath(__file__))
+    relative_path = f'{schemas_dir}/schemas/backport/{schema_name}.json'
+    path = os.path.join(dir, relative_path)
+
```

Comment:
This changes introduce duplicate code to open and parse the schema's and to create a validator. This can be avoided by putting the schemas inside the same folder as all other OCPP 1.6 schema's. Could you remove this duplication?